### PR TITLE
adding a pylint workflow

### PR DESCRIPTION
**Pull request by khyberspache**

### CODE REVIEW

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Pylint
```
Comment (privateducky) [inline .github/workflows/pylint.yml:1]: if we added another language, like JS, to the mix here.. what would that look like?

File: .github/workflows/pylint.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+    - name: Analysing the code with pylint
```
Comment (privateducky) [inline .github/workflows/pylint.yml:21]: if we made the 'dotfiles' repo public, could we clone here instead, to create consistency?

Comment (khyberspache) [reply .github/workflows/pylint.yml:21]: yep would work fine

File: .github/workflows/pylint.yml (hunk 1 above)
Comment (khyberspache) [reply .github/workflows/pylint.yml:1]: we would add either:
a) another workflow yml
b) break this into 2 jobs and run one against all the JS files

**Comment by privateducky** [comment]
closing to resume this work at some point in the future.

**Pull request closed without merging**